### DISCUSSION TITLE
Permanent Delete (softdelete)

### DISCUSSION THREAD

**Discussion by ghost** [thedevdojo/voyager#5297] [Q&A]
Please anyone can help me. I've successfully added the softdelete function and it's working amazingly. But I want to have another button next to restore to be able to permanently delete the data from DB.

<img width="880" alt="Capture" src="https://user-images.githubusercontent.com/35699210/115619541-0ccf2c80-a305-11eb-8a1b-9d69cd8343e0.PNG">

**Comment by emptynick** [comment]
Voyager has [actions buttons](https://voyager-docs.devdojo.com/customization/action-buttons) which allow you to do that.
From there you just need to call `forceDelete()` on the item.

**Reply by ghost** [reply]
Thank you so much for your reponse.

I did try but am failing to do so. Is there a tutorial or any place where i can see how to implement it plz?

**Reply by ghost** [reply]
Please anyone can assist me?

**Reply by MrCrayon** [reply]
How to create a custom button is explained in documentation https://voyager-docs.devdojo.com/customization/action-buttons 
Then I'd advice to check Voyager source code to create the right policy
https://github.com/the-control-group/voyager/blob/1.4/src/Policies/BasePolicy.php
Everything else is Laravel.
